source/plugins/go/src/utils.go

```diff
@@ -280,3 +281,30 @@ func getGenevaWindowsNamedPipeName() string {
 	}
 	return namedPipeName
 }
+
+// get the Output stream ID tag value corresponding to the datatype
+func getOutputStreamIdTag(dataType string) string {
+	useFromCache := true
+	switch dataType {
+	case ContainerLogDataType, ContainerLogV2DataType:
+		elapsed := time.Now().Sub(MdsdContainerLogTagRefreshTracker)
+		if !strings.HasPrefix(MdsdContainerLogTagName, MdsdOutputStreamIdTagPrefix) || elapsed.Seconds() >= agentConfigRefreshIntervalSeconds {
+			useFromCache = false
+			MdsdContainerLogTagRefreshTracker = time.Now()
+		}
+	case KubeMonAgentEventDataType:
```

Comment:
By creating a function, I was actually thinking passing all the values by reference for the RefreshTracker, datatype, etc. So, we just have a 5 line function that works for all the datatypes. But, it is okay I think.

Reply:
> By creating a function, I was actually thinking passing all the values by reference for the RefreshTracker, datatype, etc. So, we just have a 5 line function that works for all the datatypes. But, it is okay I think.

good idea. refactored it.